### PR TITLE
Lambda-Kinesis Data Stream-Lambda

### PR DESCRIPTION
**Pull request by sabornibhattacharya**
*Issue #, if available:*

*Description of changes:*

Adding a use case where Kinesis Data Stream can have serverless producers and consumers. The folder lambda-kinesis-lambda contains the tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lambda-kinesis-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# A demonstration of serverless producer-consumer use case using AWS Lambda and AWS Kinesis Data Stream
+
+This pattern creates an AWS Kinesis Data Stream, a stream consumer, and an AWS Lambda function. When data is added to the stream, the Lambda function is invoked.
+
+Learn more about this pattern at Serverless Land Patterns: [https://serverlessland.com/patterns/kinesis-to-lambda](https://serverlessland.com/patterns/kinesis-to-lambda)
```
Comment (mavi888) [inline lambda-kinesis-lambda/README.md:5]: right endpoint will be same as this directory - https://serverlessland.com/patterns/lambda-kinesis-lambda

File: lambda-kinesis-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# A demonstration of serverless producer-consumer use case using AWS Lambda and AWS Kinesis Data Stream
+
+This pattern creates an AWS Kinesis Data Stream, a stream consumer, and an AWS Lambda function. When data is added to the stream, the Lambda function is invoked.
```
Comment (mavi888) [inline lambda-kinesis-lambda/README.md:3]: This pattern creates 2 lambda functions, one that puts data in the stream and another that gets trigger when there is data in the stream. Please improve this description, to what the code is doing

File: lambda-kinesis-lambda/template.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  # Define a Lambda function as an application consumer
+  ConsumerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: consumer/
+      Runtime: python3.9
+      Handler: consumer-function.lambda_handler
+      Policies: 
+      - AWSLambdaKinesisExecutionRole
```
Comment (mavi888) [inline lambda-kinesis-lambda/template.yaml:40]: Can you give a more specific policy. Like https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html#kinesis-stream-read-policy ? where it giving read permissions to just the stream you want and not to all the streams in the account?

**Comment by mavi888** [comment]
@sabornibhattacharya please fix the comments and then I can merge it :)

File: lambda-kinesis-lambda/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        "key3": "value3"
+    }
+    ```
+2. Observe the logs of the consumer Lambda function to verify if the data pushed by the producer is received or not.
+
+
+## Cleanup
+
+1. Delete the stack
+    ```bash
```
Comment (mavi888) [inline lambda-kinesis-lambda/README.md:54]: for deleting the stack you can use now sam delete, please update

File: lambda-kinesis-lambda/README.md (hunk 2 above)
Comment (sabornibhattacharya) [reply lambda-kinesis-lambda/README.md:3]: Added as per suggestion

File: lambda-kinesis-lambda/template.yaml (hunk 3 above)
Comment (sabornibhattacharya) [reply lambda-kinesis-lambda/template.yaml]: Changes made as suggested.

File: lambda-kinesis-lambda/README.md (hunk 4 above)
Comment (sabornibhattacharya) [reply lambda-kinesis-lambda/README.md]: Updated.

File: lambda-kinesis-lambda/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# This pattern creates 2 Lambda functions, one that puts data into the Kinesis Data Stream and another one that gets trigger when there is data in the stream.
```
Comment (mavi888) [inline lambda-kinesis-lambda/README.md:1]: This is good, but this is not the title of the file. This is a good intro to the file after the title. 

For the title, something like you had before is good.
# Serverless producer-consumer use case using AWS Lambda and AWS Kinesis Data Stream

File: lambda-kinesis-lambda/README.md [hunk 6]
```diff
@@ -50,17 +50,8 @@ Important: this application uses various AWS services and there are costs associ
...
-    ```
-2. Confirm the stack has been deleted
-    ```bash
-    aws cloudformation list-stacks --query "StackSummaries[?contains(StackName,'STACK_NAME')].StackStatus"
-    ```
-OR
+For deleting the stack you can use now sam delete from SAM CLI -
 
-Simply run the following command to delete your stack from SAM CLI - 
     ```bash
```
Comment (mavi888) [inline lambda-kinesis-lambda/README.md:55]: this doenst show propertly in md

for that just do 
``` 
sam delete
```

**Comment by mavi888** [comment]
Can you add the [example-pattern.json](https://github.com/aws-samples/serverless-patterns/blob/main/_pattern-model/example-pattern.json) file to your pattern please? 
You can find more information in our contribution guidelines https://github.com/aws-samples/serverless-patterns/blob/main/.github/ISSUE_TEMPLATE/new-serverless-pattern-submission.md

**Comment by mavi888** [comment]
@sabornibhattacharya few more things before we can merge it :)

File: lambda-kinesis-lambda/README.md (hunk 5 above)
Comment (sabornibhattacharya) [reply lambda-kinesis-lambda/README.md:1]: Done

**Comment by sabornibhattacharya** [comment]
> 



> Can you add the [example-pattern.json](https://github.com/aws-samples/serverless-patterns/blob/main/_pattern-model/example-pattern.json) file to your pattern please? You can find more information in our contribution guidelines https://github.com/aws-samples/serverless-patterns/blob/main/.github/ISSUE_TEMPLATE/new-serverless-pattern-submission.md

Added example-pattern.json as suggested. Please let me know if anymore changes are required.

**Comment by mavi888** [comment]
Thanks for your contribution, your pattern is now live here https://serverlessland.com/patterns/lambda-kinesis-lambda

**Pull request merged**